src/SimpleMutex.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Thesis\MessageBus\Transport\Amqp;
+
+use Amp\Cancellation;
+use Amp\Future;
+use function Amp\async;
+
+/**
+ * @api
+ * @template T
+ */
+final class SimpleMutex
```

Comment:
I think it should be named [OnceValue](https://pkg.go.dev/sync#OnceValue), since this primitive is not about protecting data, but rather about ensuring exclusive initialization of a certain value.